### ISSUE TITLE
don't use modals

### ISSUE THREAD

**Issue by agarwal** [OCamlPro/ows#12]
Clicking on a package pops up a modal. I think it would be more convenient as a regular page. In particular, this would allow linking to a package's page.

**Comment by abate** [comment]
there is a direct link to a package status (ex : http://ows.irill.org/latest/today/#packagewebdav ) . Something I might consider doing is to create a page for each maintainer with the status of all his/her packages. I think modal are a quick solution to browse multiple packages at once. But I see the downside of it. It might be worth exploring the possibilities of adding a sort of permanent link to the package status that is not a modal. Patches are definitely welcome !

**Comment by agarwal** [comment]
Right, sorry I failed to notice the link target was a bookmarkable link. I just assumed it wasn't. So it's just an aesthetic preference. I can't reproduce it now, but sometimes the modal popped up out-of-view, and I'd have to scroll up to realize it is even there. There are always little problems like that with dynamic stuff, so I wonder if it is actually worth it in this case.

**Comment by dbuenzli** [comment]
FWIW I also found the modals to be quite annoying. My back button is always easier to find than looking up for the way to close the modal.

**Comment by abate** [comment]
The modal can also be closed using the "esc" button. What is clearly annoying about modals, is that to get a direct link to a package status, you need to load the entire page that can be quite heavy. I've added a simple static page now : ex http://ows.irill.org/latest/today/packages/dose-page.html

I've to clean it up, and my plan is to add a "permalink" to this page, while leaving the modal structure to browse the repository .

**Comment by dbuenzli** [comment]
> The modal can also be closed using the "esc" button.

Yes, but that's not the way I browse the www. 

> I've to clean it up, and my plan is to add a "permalink" to this page, while leaving the modal structure to browse the repository .

TBH, I don't think modals are great way of browsing the repository. See for example http://opam.ocaml.org/packages/ which is very simple yet much more convenient for browsing. Having two kind of links "modal" and "permalink" complexifies the UI for no good reason especially since I don't understand what the modal structure brings from a user experience point view except breaking both my browser history and back button.

**Comment by abate** [comment]
to have a nice modern UI, I've used js and it takes considerable time to load. This give you the table, the search and few bells and whistles. The modal allows you to avoid reloading this page while browsing the repository. Having a static page and hitting back all the time will force you to re-run the js part and that can be frustrating time wise. This is the main reason why I've used modals. Certainly, just using a plain html table will resolve the problem, but then we could just go back to the old ows. The permlink I was talking about will basically serve the people that want to exchange a ows link without having to load the entire package table. I understand you objection. I'm just trying to find a nice way to fit all the user cases ...
